### PR TITLE
Stream sync api test

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Final test for stream sync to test API functionality.

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
@pfultz2 just ended up grabbing code from hip.cpp and putting that here and still running into problems. 

replacing the call to add the input buffer via to_gpu as a program parameter yields

```
root@d5f1b0c18dc9:/code/AMDMIGraphX/build# bin/test_api_gpu load_and_run_async
[   RUN    ] load_and_run_async
"Memobj map already has an en
```

File: test/api/test_gpu.cpp [hunk 1]
```diff
@@ -45,27 +46,68 @@ TEST_CASE(load_and_run)
...
+
+    std::vector<hip_ptr> host_buffs;
+    std::vector<hip_ptr> buffs;
+    std::vector<migraphx::argument> args;
     for(auto&& name : param_shapes.names())
     {
-        pp.add(name, migraphx::argument::generate(param_shapes[name]));
+        args.push_back(migraphx::argument::generate(param_shapes[name]));
+        buffs.push_back(get_hip_buffer(args.rbegin()->get_shape().bytes()));
+        host_buffs.push_back(get_hip_buffer(args.rbegin()->get_shape().bytes()));
```
Comment (umangyadav) [inline test/api/test_gpu.cpp:100]: Doesnt' this `host_buffs` needs to point to `args` first before starting copy to GPU ?

Comment (TedThemistokleous) [reply test/api/test_gpu.cpp:100]: Ah got it. Ill fix that. Printing it out looks like I was getting all zeros.

Comment (TedThemistokleous) [reply test/api/test_gpu.cpp]: Done

**Review by umangyadav — APPROVED** [review]

**Pull request merged**